### ISSUE TITLE
False positive CWE-117 C#

### ISSUE THREAD

**Issue by mwhitney86** [github/codeql#15195]
**Description of the false positive**

C# When fixing CWE-117 with the expected fixes, if the String.Replace is done with the overload that contains the StringComparison clarifier, the fix is not recognized.

**Code samples or links to source code**

```
public async Task RequestAsync(string key)
...
_logger.log(key.Replace(Environment.NewLine, "", StringComparison.InvariantCultureIgnoreCase));
```

**URL to the alert on GitHub code scanning (optional)**

https://github.com/github/codeql/blob/d540fc0794dcb2a6c10648b8925403788612e976/csharp/ql/src/Security%20Features/CWE-117/LogForging.ql

**Comment by aeisenberg** [comment]
Thanks for reporting this false positive. This concerns a security query and we will take a look at this, though it may not be until the new year.

**Comment by michaelnebel** [comment]
It looks like we don't recognise all overloads of the replace method. I will look into this.

**Comment by michaelnebel** [comment]
This will be fixed here: https://github.com/github/codeql/pull/15212

**Comment by mwhitney86** [comment]
Thank you!

**Comment by mwhitney86** [comment]
@michaelnebel I'm not familiar with yalls release process, when will this be available to my codeql scanner? Thanks

**Comment by michaelnebel** [comment]
> [@michaelnebel](https://github.com/michaelnebel) I'm not familiar with yalls release process, when will this be available to my codeql scanner? Thanks

It should be available with CodeQL 2.16.0, which is expected to be generally available on January 12th.
